klog.go

```diff
@@ -520,6 +532,96 @@ func (l *loggingT) setVState(verbosity Level, filter []modulePat, setFilter bool
 
 var timeNow = time.Now // Stubbed out for testing.
 
+// CaptureState gathers information about all current klog settings.
+// The result can be used to restore those settings.
+func CaptureState() State {
+	logging.mu.Lock()
+	defer logging.mu.Unlock()
+	return &state{
+		// We cannot simply copy the entire struct because
+		// it contains a mutex which cannot be copied.
+		loggingT: loggingT{
```

Comment:
I wonder whether I should refactor loggingT to make capture/restore simpler.

I'm a bit afraid of breaking things, though.

Reply:
> I'm a bit afraid of breaking things, though.

Would prefer if we rephrased it to "Do we trust your tests to make a refactor?" or better "What tests should we add to feel safe?"

Reply:
I can't think of problems that might go unnoticed. Most likely it wouldn't even compile.

Okay, so let me see whether refactoring loggingT makes the PR look better.

Reply:
I've added a second commit where I move variables around.

The contextual logging variables where in the file that owned them because that seemed cleaner and I still had the hope of someday using klog in Kubernetes without depending on klog.go, but that seems unlikely, so I moved them too because they are part of the global settings that need to be captured.

Regarding tests: I don't know how to automatically test that all fields are properly deep-copied. Right now I am just testing the one field where I know that special handling is required. I could probably write something based on reflect, but then the test easily because more complex that the code itself and itself would need tests.